contracts/mocks/MockMarketplace.sol

```diff
@@ -16,20 +19,35 @@ contract MockMarketplace {
         tokenAddress.transferFrom(msg.sender, recipient, _tokenId);
     }
 
+    /// @notice This code is only for testing purposes. Do not use similar
+    /// @notice constructions in production code as they are open to attack.
+    /// @dev For details see: https://github.com/crytic/slither/wiki/Detector-Documentation#arbitrary-from-in-transferfrom
     function executeTransferFrom(address from, address to, uint256 _tokenId) public {
+        // slither-disable-next-line arbitrary-send-erc20
         tokenAddress.transferFrom(from, to, _tokenId);
     }
 
     function executeApproveForAll(address operator, bool approved) public {
         tokenAddress.setApprovalForAll(operator, approved);
     }
 
+    /// @notice This code is only for testing purposes. Do not use similar
+    /// @notice constructions in production code as they are open to attack.
+    /// @dev For details see: https://github.com/crytic/slither/wiki/Detector-Documentation#arbitrary-from-in-transferfrom
     function executeTransferRoyalties(address from, address recipient, uint256 _tokenId, uint256 price) public payable {
+        // solhint-disable-next-line custom-errors
         require(msg.value == price, "insufficient msg.value");
         (address receiver, uint256 royaltyAmount) = royaltyAddress.royaltyInfo(_tokenId, price);
         uint256 sellerAmt = msg.value - royaltyAmount;
+        if (receiver == address(0)) {
+            revert ZeroAddress();
+        }
         payable(receiver).transfer(royaltyAmount);
+        if (from == address(0)) {
+            revert ZeroAddress();
+        }
```

Comment:
i think this can be merged with the if statement above. If it needs to be separate, I'd suggest performing the validation earlier before additional work is done, since it is just validating a function arg

Reply:
Fixed